src/link/link_info/iptunnel.rs

```diff
@@ -202,13 +207,26 @@ impl<'a, T: AsRef<[u8]> + ?Sized> Parseable<NlaBuffer<&'a T>> for InfoIpTunnel {
                 parse_u32(payload)
                     .context("invalid IFLA_IPTUN_FLOWINFO value")?,
             ),
-            IFLA_IPTUN_FLAGS => Flags(TunnelFlags::from_bits_retain(
-                parse_u32(payload).context("invalid IFLA_IPTUN_FLAGS value")?,
-            )),
-            IFLA_IPTUN_PROTO => Protocol(IpProtocol::from(
-                parse_u8(payload).context("invalid IFLA_IPTUN_PROTO value")?
-                    as i32,
-            )),
+            IFLA_IPTUN_FLAGS => match kind {
+                super::InfoKind::SitTun => {
+                    // Parse as u16 for SIT tunnels
+                    Flags(TunnelFlags::from_bits_retain(u32::from(
+                        parse_u16(payload).context(
+                            "invalid IFLA_IPTUN_FLAGS value for SIT",
+                        )?,
```

Comment:
Instead of storing u16 into u32.

I prefer `InfoIpTunnel::Ipv6SitFlags(u16)`, `InfoIpTunnel::Ipv6Flags(u32)`, `InfoIpTunnel::Ipv4Flags(u16)`.